core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala

```diff
@@ -73,7 +73,7 @@ private[spark] class MetricsSystem private (
   private[this] val metricsConfig = new MetricsConfig(conf)
 
   private val sinks = new mutable.ArrayBuffer[Sink]
-  private val sources = new mutable.ArrayBuffer[Source]
+  val sources = new mutable.ArrayBuffer[Source]
```

Comment:
I think it would be better not to loose the restriction to expose this to the public user just only for test convenience. you could refer [here](https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/ExecutorAllocationManagerSuite.scala#L782) to see how to test with private field.